File: 프로그래머스/1/42748. K번째수/K번째수.cpp
#include <iostream>
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> array, vector<vector<int>> commands) {

    vector<int> answer;

    // for ( int i = 0 ; i < commands.size(); i++) 도 가능
    for (auto i : commands)
    {
        vector<int> temp;

        for (int j = 0; j <= i[1]-i[0]; j++)
        {
            temp.push_back(array[i[0]-1+j]);
        }
        sort(temp.begin(), temp.end());
        answer.push_back(temp[i[2]-1]);

    }


    return answer;
}

File: 프로그래머스/1/42748. K번째수/K번째수.cs
using System;
using System.Collections.Generic;

public class Solution {
    public int[] solution(int[] array, int[,] commands) {
        int[] answer = new int[commands.GetLength(0)];
        List<int> list = new List<int>();
        for (int i = 0; i < commands.GetLength(0) ; i++)
        {
            int start = commands[i,0];
            int end = commands[i,1];
            int num = commands[i,2];
            
            for (int j = start-1; j <= end-1; j ++ )
            {
                list.Add(array[j]);
            }
            
            // foreach ( int z in list )
            // {
            //     Console.WriteLine(z);
            // }
            // Console.WriteLine("end");
            
            list.Sort();
            answer[i] = list[num-1];
            list.Clear();

        }
        
        return answer;
    }
}